File: SYP.Web/Modules/ServerTypes/Warehouse/StockMovementsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StockMovementsRow } from "./StockMovementsRow";

export interface StockMovementsColumns {
    MovementType: Column<StockMovementsRow>;
    DocumentNo: Column<StockMovementsRow>;
    MovementDate: Column<StockMovementsRow>;
    WarehouseName: Column<StockMovementsRow>;
    ProductCode: Column<StockMovementsRow>;
    ProductName: Column<StockMovementsRow>;
    Quantity: Column<StockMovementsRow>;
    Status: Column<StockMovementsRow>;
    Description: Column<StockMovementsRow>;
}

export class StockMovementsColumns extends ColumnsBase<StockMovementsRow> {
    static readonly columnsKey = 'Warehouse.StockMovements';
    static readonly Fields = fieldsProxy<StockMovementsColumns>();
}

File: SYP.Web/Modules/Warehouse/StockMovements/StockMovementsColumns.cs
namespace SYP.Warehouse.Columns;

[ColumnsScript("Warehouse.StockMovements")]
[BasedOnRow(typeof(StockMovementsRow), CheckNames = true)]
public class StockMovementsColumns
{
    [Width(80)]
    public string MovementType { get; set; }

    [Width(120)]
    public string DocumentNo { get; set; }

    [Width(150), DisplayFormat("dd.MM.yyyy HH:mm")]
    public DateTime MovementDate { get; set; }

    [Width(150)]
    public string WarehouseName { get; set; }

    [Width(100)]
    public string ProductCode { get; set; }

    [Width(200)]
    public string ProductName { get; set; }

    [Width(100), AlignRight]
    public decimal Quantity { get; set; }

    [Width(80)]
    public short Status { get; set; }

    [Width(200)]
    public string Description { get; set; }
}
